### PR TITLE
CI: add compile typescript step

### PR DESCRIPTION
**Pull request by webwarrior-ws**
Add "compile typescript" step to CI so that any compiler errors in tests are reported rather than lead to test failure with unclear cause.

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -99,6 +99,11 @@ jobs:
           npm --version
           npx commitlint --version
           yarn --version
+          npx tsc --version
+      - name: Run typescript compiler
+        # This step is needed because jest will swallow any compiler error
```
Comment (knocte) [inline .github/workflows/CI.yml:104]: @webwarrior-ws shouldn't this say "would" instead of "will"?

Comment (webwarrior-ws) [reply .github/workflows/CI.yml:104]: It will swallow a compiler error, if there is any.

Comment (knocte) [reply .github/workflows/CI.yml:104]: yes but after this commit it will not; so putting this comment is misleading, change the "will" with "would" please

Comment (webwarrior-ws) [reply .github/workflows/CI.yml]: Done.

**Comment by knocte** [comment]
I just compared the compiler errors between the 1st and 2nd commit and I don't see any differences; so is the commit title "CI: set typescript version to 4.9" really needed at all?

**Comment by webwarrior-ws** [comment]
> I just compared the compiler errors between the 1st and 2nd commit and I don't see any differences; so is the commit title "CI: set typescript version to 4.9" really needed at all?

You're right, this step is not needed. Removed that commit.

**Comment by knocte** [comment]
Please split the 1st commit in two. One that simply adds the compilation, and a 2nd commit that explains the downleveliteration thing (also, this latter setting can be set in the json file rather than an argument).

**Pull request merged**